examples/cloudml-churn-prediction/preprocessor/preprocessor/preprocess.py

```diff
@@ -0,0 +1,325 @@
+            elif r < 1 - train_size:
+                yield beam.pvalue.TaggedOutput('Val', element)
+            else:
+                yield element
+
+    split_data = (
+        p | 'SplitData' >> beam.ParDo(_SplitData()).with_outputs(
+            'Test',
+            'Val',
+            main='Train'))
+    return split_data['Train'], split_data['Val'], split_data['Test']
+
+
+def parse_arguments(argv):
+    """Parse command-line arguments."""
+    parser = argparse.ArgumentParser()
+    timestamp = datetime.now().strftime('%Y%m%d%H%M%S')
+    parser.add_argument(
+        '--job_name',
+        default='{}-{}'.format('bq-to-tfrecords', timestamp)
+    )
+    parser.add_argument(
+        '--output_dir',
+        required=True,
+    )
+    parser.add_argument(
+        '--log_level',
+        help='Set logging level',
+        default='INFO'
+    )
+    parser.add_argument(
+        '--bq_table',
+        help="""Source BigQuery table in [Project]:[Dataset]:[Table]
+            format.""",
+        default='bigquery-public-data.google_analytics_sample.ga_sessions_*'
+    )
+    parser.add_argument(
+        '--machine_type',
+        help="""Set machine type for Dataflow worker machines.""",
+        default='n1-highmem-4'
+    )
+    parser.add_argument(
+        '--cloud',
+        help="""Run preprocessing on the cloud. Default False.""",
+        action='store_true',
+        default=False,
+    )
+    parser.add_argument(
+        '--project_id',
+        help="""Google Cloud project ID""",
+        required=True,
+    )
+    known_args, _ = parser.parse_known_args(argv)
+
+    return known_args
+
+
+def get_pipeline_args(flags):
+    """Create Apache Beam pipeline arguments"""
+    options = {
+        'project': flags.project_id,
+        'staging_location': os.path.join(flags.output_dir, 'staging'),
+        'temp_location': os.path.join(flags.output_dir, 'temp2'),
+        'job_name': flags.job_name,
+        'save_main_session': True,
+        'setup_file': './setup.py'
+    }
+    return options
+
+
+def run():
+    """Run Apache Beam pipeline to generate TFRecords for Survival Analysis"""
+    flags = parse_arguments(sys.argv[1:])
+    pipeline_args = get_pipeline_args(flags)
+
+    options = PipelineOptions(flags=[], **pipeline_args)
+    options.view_as(WorkerOptions).machine_type = flags.machine_type
+
+    input_spec = features.RAW_FEATURE_SPEC
+    temp_dir = os.path.join(flags.output_dir, 'tmp')
+
+    runner = 'DataflowRunner' if flags.cloud else 'DirectRunner'
+
+    with beam.Pipeline(runner, options=options) as p:
+        with tft_beam.Context(temp_dir=temp_dir):
+            raw_data = (p 
+                        | 'QueryTable' >> beam.io.Read(beam.io.BigQuerySource(
+                            query=query.get_query(flags.bq_table),
+                            project=flags.project_id,
+                            use_standard_sql=True))
+                        #omit 'Generate Data' step if working with real data
```

Comment:
Decided not to have a generate_data flag, because the pipeline will not run if these fields are not added. The pipeline requires that start_date, end_date, and active are part of each entry. This note is included in the README.md